### ISSUE TITLE
Missing async prefix for 'auth_keypair.authenticate' method

### ISSUE THREAD

**Issue by almulalic** [snowflakedb/snowflake-connector-nodejs#517]
1. What version of NodeJS driver are you using?
v19.8.1
v14.19.1
   
2. What operating system and processor architecture are you using?
Macbook 2019 - MacOS - Ventura 13.3.1
   
3. What version of NodeJS are you using?
v14.19.1 and 6.14.16

4. What are the component versions in the environment (`npm list`)?
Too big to attach and not relevant to the problem. Only thing that I have installed is the SDK.

5. Server version:* E.g. 1.90.1
7.18.1

6. What did you do?
Tried to initalize snowflake connector with `SNOWFLAKE_JDBC` connection. I passed the same arguments as I passed to the `snowsql` cmd util but it failed on a weird NodeJS error. I tried to run it with multiple versions of node (19, 16, 14) and both typescript and javascript but nothing seemed to work. Finally I started digging through the connection files and debugging and I found that one of your authentication methods is missing `async` prefix because the function that calls it expects a promise-like interface. I fixed it by adding async prefix (function doesen't need to be async) but this can also be fixed if the caller function could identify if the return type is resolved to promise or not. See the [PR](https://github.com/snowflakedb/snowflake-connector-nodejs/pull/518) for more info.

7. What did you expect to see?
Executed Snowflake query or a valid error message.

8. Can you set logging to DEBUG and collect the logs?
Logs are self explanatory, it is failing becuase `auth.authenticate` is expected to be asynchronus function that provides the .then, .catch interface.

```
╰─ ts-node main.ts                                                                                                                                                  ─╯
(node:2876) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'then' of undefined
    at Connection.connectAsync (/path/to//project/test_snowflake_conn/node_modules/snowflake-sdk/lib/connection/connection.js:298:9)
    at /path/to//project/test_snowflake_conn/main.ts:18:39
    at Generator.next (<anonymous>)
    at /path/to//project/test_snowflake_conn/node_modules/tslib/tslib.js:167:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/path/to//project/test_snowflake_conn/node_modules/tslib/tslib.js:163:16)
    at testConnection (/path/to//project/test_snowflake_conn/main.ts:6:20)
    at /path/to//project/test_snowflake_conn/main.ts:33:9
    at Generator.next (<anonymous>)
    at /path/to//project/elasticdump-tool/test_snowflake_conn/node_modules/tslib/tslib.js:167:75
(Use `node --trace-warnings ...` to show where the warning was created)
(node:2876) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:2876) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:2876) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.

Please migrate your code to use AWS SDK for JavaScript (v3).
For more information, check the migration guide at https://a.co/7PzMCcy
```

This call expects async:
<img width="730" alt="Screenshot 2023-06-01 at 21 44 45" src="https://github.com/snowflakedb/snowflake-connector-nodejs/assets/47641522/d7ba6838-5d03-4f45-ba9d-8833aab32e58">

This returns nothing:
<img width="816" alt="Screenshot 2023-06-01 at 21 45 16" src="https://github.com/snowflakedb/snowflake-connector-nodejs/assets/47641522/1fa323a7-1ac7-4f13-855d-d3a4105b58f9">

10. What is your Snowflake account identifier, if any? (Optional)
- NDA

Minimal reproducable setup (ts):
```
import snowflake from "snowflake-sdk"

async function testConnection() {
  
  const connection = snowflake.createConnection({
    account: "account",
    username: "username",
    role: "role",
    database: "database",
    schema: "schema",
    warehouse: "warehouse",
    authenticator: "SNOWFLAKE_JWT",
    privateKeyPath: "privateKeyPath",
    privateKeyPass: "privateKeyPass",
  });
  
  const statement = (await connection.connectAsync()).execute({
    sqlText: "SHOW GRANTS;",
    complete: function (err, stmt, rows) {
      if (err) {
        console.error("Failed to execute statement due to the following error: " + err.message);
      } else {
        console.log("Successfully executed statement: " + stmt.getSqlText());
        console.log(rows);
      }
    },
  });
}

(async() => {
  await testConnection()
})()

```

**Comment by sfc-gh-dszmolka** [comment]
hi and thank you for raising this issue ! looks similar to https://github.com/snowflakedb/snowflake-connector-nodejs/issues/308 and obviously keeps causing headache, so thank you for the PR too ! we'll take a look

**Comment by sfc-gh-dszmolka** [comment]
#522

**Comment by sfc-gh-dszmolka** [comment]
fix is out with release 1.6.23
